src/status_im2/contexts/chat/messages/pin/list/view.cljs

```diff
@@ -1,37 +1,56 @@
 (ns status-im2.contexts.chat.messages.pin.list.view
-  (:require [utils.i18n :as i18n]
-            [quo2.core :as quo]
+  (:require [quo2.core :as quo]
             [quo2.foundations.colors :as colors]
             [react-native.core :as rn]
             [status-im2.contexts.chat.messages.content.view :as message]
-            [utils.re-frame :as rf]
-            [utils.datetime :as datetime]))
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
 
 (def list-key-fn #(or (:message-id %) (:value %)))
 
+(defn get-render-data
+  [{:keys [group-chat chat-id public? community-id admins space-keeper edit-enabled
+           in-pinned-view?]}]
+  (let [current-public-key (rf/sub [:multiaccount/public-key])
+        {:keys [can-delete-message-for-everyone-in-community?] :as community} (rf/sub
+                                                                               [:communities/community
+                                                                                community-id])
+        group-admin? (get admins current-public-key)
+        community-admin? (when community (community :admin))
+        message-pin-enabled (and (not public?)
+                                 (or (not group-chat)
+                                     (and group-chat
+                                          (or group-admin?
+                                              community-admin?))))]
+    {:group-chat                                    group-chat
+     :public?                                       public?
+     :community?                                    (not (nil? community-id))
+     :group-admin?                                  group-admin?
+     :community-admin?                              community-admin?
+     :current-public-key                            current-public-key
+     :space-keeper                                  space-keeper
+     :chat-id                                       chat-id
+     :message-pin-enabled                           message-pin-enabled
+     :edit-enabled                                  edit-enabled
+     :in-pinned-view?                               in-pinned-view?
+     :can-delete-message-for-everyone-in-community? can-delete-message-for-everyone-in-community?}))
```

Comment:
copied from `status-im2.contexts.chat.messages.list.view`
not sure is it better to copy or reuse
https://github.com/status-im/status-mobile/blob/def02d48dabc7996b9a4d7ffd1f28fbf8ff46838/src/status_im2/contexts/chat/messages/list/view.cljs#L71

Reply:
better to move it and reuse

Reply:
fixed by reuse the the one in `status-im2.contexts.chat.messages.list.view`
added doc string `"compute data used to render message list, including pinned message list and message list in chats"` for `get-render-data`